### ISSUE TITLE
tRecX django app - revival

### ISSUE THREAD

**Issue by scrinzi** [GT-Nexus/AMOS-Gateway#14]
Django App Issues

on dev.ampgateway.org

- plot panel does not plot existing "plotables": error importing matplotlib for plot python
- plot panel cannot be enlarged beyond rather small area
- does not submit (remains status "unsubmitted")

on amosgateway.org:

- machines in list: expanse and stampede2 (I know that installation on the Fujitsu machine was done)
- plotting works OK
- plot panel resize limited to too small a range
- executes on expanse and stampede2
- returns status COMPLETED on expanse
- returns status FAILED on stampede2

- retrieval of files fails on both machines with message like:

Network Error Unknown error when fetching viewables.
2d83/t.exports@https://amosgateway.org/static/trecx_django_app/dashboard/dist/js/chunk-vendors.a4a31b01.js:31:8163
467f/t.exports@https://amosgateway.org/static/trecx_django_app/dashboard/dist/js/chunk-vendors.a4a31b01.js:37:26842
b50d/t.exports/</f.onreadystatechange@https://amosgateway.org/static/trecx_django_app/dashboard/dist/js/chunk-vendors.a4a31b01.js:59:36844

Needed steps: 

- fix file retrieval
- if dev.ampgateway is to continue use, fix submit issues there
- install on more resources (e.g. make the Fujitsu machine available, code runs there)

**Comment by scrinzi** [comment]
Follow up: 
if there is a chance, it would be best to get the DEV interface up first, as, if that works, chances are I can implement the new features and that gets pushed to AMOS then more quickly.

**Comment by spamidig** [comment]
Armin: 
I will look into this in the next few days..

**Comment by spamidig** [comment]
Now the plot is back and I have corrected the fftw issue on Expanse and Expanse runs are fine. The expanded plot panel requires code changes and we will look into that probably in January.

**Comment by scrinzi** [comment]
Verified clone/edit/compute/plot cycle - works correctly on stampede (through dev.amp)

**Comment by scrinzi** [comment]
the initial issue with file transfer has re-surfaced, see attachment
![django_failure](https://github.com/GT-ARTISAN/AMOS-Gateway/assets/2103479/196aa4ce-c831-4249-ad10-45007feab74a)

**Comment by spamidig** [comment]
Armin:

Is this a recent run. Can you provide the jobid/expt ID details from the workspace/Expt Statistics pages for the run.

Thanks,
Sudhakar.

From: scrinzi ***@***.***>
Date: Monday, January 8, 2024 at 10:32 AM
To: GT-ARTISAN/AMOS-Gateway ***@***.***>
Cc: Pamidighantam, Sudhakar ***@***.***>, Assign ***@***.***>
Subject: Re: [GT-ARTISAN/AMOS-Gateway] tRecX django app - revival (Issue #14)

the initial issue with file transfer has re-surfaced, see attachment
django_failure.png (view on web)<https://github.com/GT-ARTISAN/AMOS-Gateway/assets/2103479/196aa4ce-c831-4249-ad10-45007feab74a>

**Comment by scrinzi** [comment]
Yes, it is the most recent run in "WorksAgain/0017" as shown on the statistics page.
0016 seems to have failed in a similar fashen
0015 worked

**Comment by scrinzi** [comment]
back working again - was there an intervention or was it a temporary problem?

**Comment by spamidig** [comment]
I changed the modules loaded.. some shared objects were not resolved perhaps due to some changes on the system.

This means that we need to run automated tests perhaps daily to make sure we catch anything that has changed...

Thanks,
Sudhakar.
________________________________
